### PR TITLE
B 2 testing

### LINKED ISSUE #43: Backend User Story 2
#31 - User Story 2: User update information can be successful

1. userUpdate mutation (can reuse productUpdate code)

1. update type.js and add:
> - [x] shipping address - break it down into components (add 1, add 2, city, state, zip)
> - [x] email
> - [x] description

1. update model.js
> - [x] adddress & it's components as STRING
> - [x] description - TEXT
> - [x] email STRING

1. migration
> - [x] need to add shipping address components to user table and description

1. resolver.js
> - [x] need to create update function (take from product update in resolver)

1. testing
- [x] completed && at 100%
- Some setup instructions for Jest/Supertest [here](https://docs.google.com/document/d/1WP1yHWMnTHV6WWimB5CNNuPStlUgdnY3Akr5zSDJGRU/edit)

### LINKED ISSUE #45: Backend User Story 3
#32 - User Story 3: User can upload a photo

1. userUpdate mutation (can reuse productUpdate code)

1. update type.js and add:
> - [x] image needs to be added as a field to usertype

1. update model.js
> - [x] image is STRING

1. migration
> - [x] need to add image column to user table

1. resolver.js
> - [x] need to add to update function (take from product update in resolver)

### LINKED ISSUE #46: Backend User Story 4
#33  - User Story 4: User can select next delivery date

1. update type.js and add:
> - [ ] add delivery date to user - enum 1 or 2 for 1st or 15th (reuse code from 

1. update model.js
> - [ ] deliveryDate datatype = Integer

1. migration
> - [ ] add deliveryDate to migration (try to reuse code there) 

1. params.json (in config folder)
This is where the enum relations are made
- [ ] need to add deliveryDate to this matching the json format mappin out 1 to 1st and 2 to 15th

1. resolver.js
> - [ ] reuse getType method in Product Types & call it getDeliveryDate this will be used in updateUser mutation

1. userUpdate mutation (can reuse productUpdate code)
- [ ] use getDeliveryDate to this method

### PR DESCRIPTION
**Pull request by kathytbui**
#### What's this PR do?
This branch merges in testing for the backend ensuring the correct API endpoints are available for a user's info (email, address, description, image, deliveryDate NOTE: profile-photo-rebase originally pulled in some of these changes prior to merge)

#### Where should the reviewer start?
migrations/user.js - highlights the new attribute added in all files

#### How should this be manually tested?
No

#### Any background context you want to provide?
The use is now able to update their email, address, description, image, deliveryDate in the database

#### What are the relevant tickets?
#43  #45 #46  

### Things to do after pulling down main:
run npm db:reset (to drop, create, migrate and reseed database)
run npm test:reset (only applicable if you want to run tests - need to make sure your testing environment is already configured)

### CODE REVIEW

File: code/api/src/modules/user/model.js [hunk 1]
```diff
@@ -36,6 +37,10 @@ module.exports = function(sequelize, DataTypes) {
     description: {
       type: DataTypes.STRING
     },
+    deliveryDate: {
+      type: DataTypes.STRING,
+      defaultValue: '1st'
```
Comment (dtsehl) [inline code/api/src/modules/user/model.js:42]: This makes it so if no value is given, the default value when creating a user is to have their delivery date set to the 1st every month.

File: code/api/src/tests/modules/user/mutations.test.js [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    const userID = tokenResponse.body.data.userLogin.user.id
+
+    const updateResponse = await request(server)
+      .post('/')
+      .set('Content-Type', 'application/json')
+      .set('Accept', 'application/json')
+      .send({ query: `mutation { userUpdate(id: ${userID}, deliveryDate: "15th" ) { id email streetAddress1 streetAddress2 city state zip description image deliveryDate } }` })
+      .expect(200)
+    expect(updateResponse.body.data.userUpdate.deliveryDate).toEqual('15th')
+  })
```
Comment (dtsehl) [inline code/api/src/tests/modules/user/mutations.test.js:148]: Great job with these thorough tests.

**Review by dtsehl — APPROVED** [review]

File: code/api/src/tests/modules/user/mutations.test.js (hunk 2 above)
Comment (leighlars) [reply code/api/src/tests/modules/user/mutations.test.js:148]: Thank you for getting these done so quickly.

File: code/api/src/modules/user/model.js (hunk 1 above)
Comment (leighlars) [reply code/api/src/modules/user/model.js:42]: Great, that's how it'll render on our end

**Comment by ajtran303** [comment]
Looks good here

**Pull request merged**